src/QmlControls/MissionItemMapVisual.qml

```diff
@@ -23,26 +23,34 @@ Item {
 
     property var map        ///< Map control to place item in
     property var vehicle    ///< Vehicle associated with this item
-    property var interactive: true    ///< Vehicle associated with this item
+    property bool interactive: true    ///< Vehicle associated with this item
 
     signal clicked(int sequenceNumber)
 
-    property var _visualItem
+    Loader {
+        id: mapVisualLoader
 
-    Component.onCompleted: {
-        if (object.mapVisualQML) {
-            var component = Qt.createComponent(object.mapVisualQML)
-            if (component.status === Component.Error) {
-                console.log("Error loading Qml: ", object.mapVisualQML, component.errorString())
-            }
-            _visualItem = component.createObject(map, { "map": _root.map, vehicle: _root.vehicle, 'opacity': Qt.binding(function() { return _root.opacity }), 'interactive': Qt.binding(function() { return _root.interactive }) })
-            _visualItem.clicked.connect(_root.clicked)
+        asynchronous: true
+
+        Component.onCompleted: {
```

Comment:
Is it still necessary to do this onCompleted as opposed to setting props directly? With the original code the onCompleted was needed in order to trigger the code which creates the visual item object. But with a Loader you don't need that trigger you can just set props and let it do the work. Unless I'm missing something here...

Reply:
Hmm, I guess you still need to jam in the map/vehicle and so forth props to the inner object.

Reply:
> So the properties would be uninitialized if set in the onLoaded.

Yeah, I didn't initially pick up on that which is why I was wondering why onCompleted was still needed.